File: ebook/lib/models/book_model.dart
class BookModel {
  String id;
  String title;
  String author;
  String description;
  String cover;
  String file;
  bool isTrending;

  BookModel({
    required this.id,
    required this.title,
    required this.author,
    required this.description,
    required this.cover,
    required this.file,
    required this.isTrending,
  });

  factory BookModel.fromJson(Map<String, dynamic> json) => BookModel(
        id: json["_id"], // from MongoDB
        title: json["title"],
        author: json["author"],
        description: json["description"],
        cover: json["cover"], // now full URL from Node.js
        file: json["file"],
        isTrending: json["isTrending"] ?? false,
      );

  Map<String, dynamic> toJson() => {
        "_id": id,
        "title": title,
        "author": author,
        "description": description,
        "cover": cover,
        "file": file,
        "isTrending": isTrending,
      };
}

File: Ebook-back/model/book.model.js
const mongoose = require("mongoose");

const bookSchema = new mongoose.Schema({
    title: {
        type: String,
        required: [true, "Book title is required"]
    },
    author: {
        type: String,
        required: [true, "Author name is required"]
    },
    description: {
        type: String,
        required: [true, "Description is required"]
    },
    cover: {
        type: String, // URL or path to image
        required: [true, "Cover image is required"]
    },
    file: {
        type: String, // URL or path to PDF/EPUB file
        required: [true, "Book file is required"]
    },
    isTrending: {
        type: Boolean,
        default: false
    }
}, { timestamps: true });

module.exports = mongoose.model("Book", bookSchema);
